makefile

```diff
@@ -70,42 +103,44 @@ install: libbsc.a bsc
 	chmod a+r $(DESTDIR)$(PREFIX)/include/libbsc.h
 	cp -f libbsc.a $(DESTDIR)$(PREFIX)/lib
 	chmod a+r $(DESTDIR)$(PREFIX)/lib/libbsc.a
+	cp -f libbsc.$(LIBEXT) $(DESTDIR)$(PREFIX)/lib
+	chmod a+r $(DESTDIR)$(PREFIX)/lib/libbsc.$(LIBEXT)
 
 clean:
-	rm -f *.o libbsc.a bsc
+	rm -rf *.o libbsc.a libbsc.$(LIBEXT) bsc bsc.dSYM
 
 adler32.o: libbsc/adler32/adler32.cpp
-	$(CC) $(CFLAGS) -c libbsc/adler32/adler32.cpp
+	$(CXX) $(CFLAGS) -c libbsc/adler32/adler32.cpp
 
 libsais.o: libbsc/bwt/libsais/libsais.c
 	$(CC) $(CFLAGS) -c libbsc/bwt/libsais/libsais.c
 
 coder.o: libbsc/coder/coder.cpp
-	$(CC) $(CFLAGS) -c libbsc/coder/coder.cpp
+	$(CXX) $(CFLAGS) -c libbsc/coder/coder.cpp
 
 qlfc.o: libbsc/coder/qlfc/qlfc.cpp
-	$(CC) $(CFLAGS) -c libbsc/coder/qlfc/qlfc.cpp
+	$(CXX) $(CFLAGS) -c libbsc/coder/qlfc/qlfc.cpp
 
 qlfc_model.o: libbsc/coder/qlfc/qlfc_model.cpp
-	$(CC) $(CFLAGS) -c libbsc/coder/qlfc/qlfc_model.cpp
+	$(CXX) $(CFLAGS) -c libbsc/coder/qlfc/qlfc_model.cpp
 
 detectors.o: libbsc/filters/detectors.cpp
-	$(CC) $(CFLAGS) -c libbsc/filters/detectors.cpp
+	$(CXX) $(CFLAGS) -c libbsc/filters/detectors.cpp
 
 preprocessing.o: libbsc/filters/preprocessing.cpp
-	$(CC) $(CFLAGS) -c libbsc/filters/preprocessing.cpp
+	$(CXX) $(CFLAGS) -c libbsc/filters/preprocessing.cpp
 
 libbsc.o: libbsc/libbsc/libbsc.cpp
-	$(CC) $(CFLAGS) -c libbsc/libbsc/libbsc.cpp
+	$(CXX) $(CFLAGS) -c libbsc/libbsc/libbsc.cpp
 
 lzp.o: libbsc/lzp/lzp.cpp
-	$(CC) $(CFLAGS) -c libbsc/lzp/lzp.cpp
+	$(CXX) $(CFLAGS) -c libbsc/lzp/lzp.cpp
 
 platform.o: libbsc/platform/platform.cpp
-	$(CC) $(CFLAGS) -c libbsc/platform/platform.cpp
+	$(CXX) $(CFLAGS) -c libbsc/platform/platform.cpp
 
 st.o: libbsc/st/st.cpp
-	$(CC) $(CFLAGS) -c libbsc/st/st.cpp
+	$(CXX) $(CFLAGS) -c libbsc/st/st.cpp
 
 bwt.o: libbsc/bwt/bwt.cpp
-	$(CC) $(CFLAGS) -c libbsc/bwt/bwt.cpp
+	$(CXX) $(CFLAGS) -c libbsc/bwt/bwt.cpp
```

Comment:
There seems to be an awful lot of code duplication here. Typically there is a single rule that arranges for all source code files (of one language) to be compiled into objects.